ui/src/components/ConfigurationForm/ConfigurationForm.consts.ts

```diff
@@ -87,15 +91,15 @@ const valueGetter = (
   option: FormSelectOption | FormRadioOption
 ): FormSelectOption['value'] | FormRadioOption['value'] => option.value;
 
-const getEffectImplementationFieldMessage = (effectType: EffectType): string =>
-  `Server implementation must be one of the following values: ${getEffectImplementationOptions(effectType)
+const getEffectImplementationFieldMessage = (effectType: EffectType, scalaVer: ScalaVersion): string =>
+  `Server implementation must be one of the following values: ${getEffectImplementationOptions(effectType, scalaVer)
     .map(labelGetter)
     .join(', ')}`;
 
 const REQUIRED_FIELD_MESSAGE = 'This field is required';
 
-export const createStarterValidationSchema = (isScalaVersionFieldVisible: boolean) => {
-  const baseSchema = yup
+export const createStarterValidationSchema = () => {
```

Comment:
This no longer takes parameters, we can safely turn it into a static object.

```typescript
const starterValidationSchema = {
  ...
};
```

Reply:
done